netty/src/test/java/io/grpc/netty/NettyClientTransportTest.java

```diff
@@ -276,6 +278,68 @@ public void bufferedStreamsShouldBeClosedWhenConnectionTerminates() throws Excep
     }
   }
 
+  public static class CantConstructChannel extends NioSocketChannel {
+    /** Constructor. It doesn't work. Feel free to try. But it doesn't work. */
+    public CantConstructChannel() {
+      // Use an Error because we've seen cases of channels failing to construct due to classloading
+      // problems (like mixing different versions of Netty), and those involve Errors.
+      throw new CantConstructChannelError();
+    }
+  }
+
+  private static class CantConstructChannelError extends Error {}
+
+  @Test
+  public void failingToConstructChannelShouldFailGracefully() throws Exception {
+    address = TestUtils.testServerAddress(12345);
+    authority = GrpcUtil.authorityFromHostAndPort(address.getHostString(), address.getPort());
+    NettyClientTransport transport = new NettyClientTransport(
+        address, CantConstructChannel.class, new HashMap<ChannelOption<?>, Object>(), group,
+        newNegotiator(), DEFAULT_WINDOW_SIZE, DEFAULT_MAX_MESSAGE_SIZE,
+        GrpcUtil.DEFAULT_MAX_HEADER_LIST_SIZE, authority, null);
+    transports.add(transport);
+
+    // Should not throw
+    callMeMaybe(transport.start(clientTransportListener));
+
+    // And RPCs and PINGs should fail cleanly, reporting the failure
+    Rpc rpc = new Rpc(transport);
+    try {
+      rpc.waitForResponse();
+      fail("Expected exception");
+    } catch (Exception ex) {
+      if (!(getRootCause(ex) instanceof CantConstructChannelError)) {
```

Comment:
I chose the `AssertionError` because it reports what `ex` actually is on failure. Instead of `assertTrue()` that would just say "Could not find expected error".